pioneer/packages/joy-utils/src/transport/tokenomics.ts

```diff
@@ -341,4 +350,125 @@ export default class TokenomicsTransport extends BaseTransport {
+    const numberOfForumPosts = (await this.api.query.forum.nextPostId() as PostId).toNumber();
+    const [councilMintId, contentCuratorMintId, storageMintId] = await Promise.all([await this.api.query.council.councilMint() as MintId, await this.api.query.contentWorkingGroup.mint() as MintId, await this.api.query.storageWorkingGroup.mint() as MintId]);
+    const [councilMint, contentMint, storageMint] = (await this.api.query.minting.mints.multi<Mint>([councilMintId, contentCuratorMintId, storageMintId]));
+
+    return {
+      blockHeight,
+      numberOfMembers,
+      content: content.length,
+      numberOfChannels,
+      proposalCount,
+      historicalProposals: HISTORICAL_PROPOSALS.length,
+      numberOfForumCategories,
+      numberOfForumPosts,
+      councilMintCapacity: councilMint.capacity.toNumber(),
+      councilMintSpent: councilMint.total_minted.toNumber(),
+      contentCuratorMintCapacity: contentMint.capacity.toNumber(),
+      contentCuratorMintSpent: contentMint.total_minted.toNumber(),
+      storageProviderMintCapacity: storageMint.capacity.toNumber(),
+      storageProviderMintSpent: storageMint.total_minted.toNumber()
+    };
+  }
+
+  async proposalStatistics () {
+    const proposal = {
+      all: 0,
+      Active: 0,
+      Approved: 0,
+      Rejected: 0,
+      Expired: 0,
+      Slashed: 0,
+      Canceled: 0,
+      Vetoed: 0
+    };
+
+    const returnData = {
+      text: { ...proposal },
+      spending: { ...proposal },
+      workingGroups: { ...proposal },
+      networkChanges: { ...proposal },
+      all: { ...proposal }
+    };
+
+    const proposals = await this.entriesByIds<ProposalId, ProposalDetails>(this.api.query.proposalsCodex.proposalDetailsByProposalId);
+    const proposalData = await this.api.query.proposalsEngine.proposals.multi<Proposal>(proposals.map(([id, _]) => id));
+
+    proposalData.map((proposal, index) => {
+      const proposalType = proposals[index][1].type;
+      let definedProposalType: 'text' | 'spending' | 'networkChanges' | 'workingGroups';
+      const proposalStatus = proposal.status.isOfType('Active') ? 'Active' as const : proposal.status.asType('Finalized').proposalStatus.type;
+
+      if (proposalType === 'Text') {
+        definedProposalType = 'text';
+      } else if (proposalType === 'Spending') {
+        definedProposalType = 'spending';
+      } else if (proposalType === 'RuntimeUpgrade' || proposalType === 'SetElectionParameters' || proposalType === 'SetValidatorCount') {
+        definedProposalType = 'networkChanges';
+      } else {
+        definedProposalType = 'workingGroups';
+      }
+
+      returnData[definedProposalType].all++;
+      returnData[definedProposalType][proposalStatus]++;
+      returnData.all.all++;
+      returnData.all[proposalStatus]++;
+    });
+
+    return returnData;
+  }
+
+  historicalProposalStatistics () {
+    const proposal = {
```

Comment:
I'm guessing that's the duplicate code that you mentioned in PR description.
I understand this is to prevent issues in case proposal module changes in the runtime?